homeassistant/components/ssdp/__init__.py

```diff
@@ -227,6 +228,14 @@ def async_scan(self, *_: Any) -> None:
         """Scan for new entries."""
         for listener in self._ssdp_listeners:
             listener.async_search()
+            try:
```

Comment:
We should probably trigger a broadcast send right after doing `listener.async_start` in `async_start` since otherwise the first broadcast send won't happen until the timer hits

Reply:
I think you're right, `listener.async_start` triggers a search, but the broadcas-searcht is "manual".

Reply:
Added an additional search. Not sure if this is what you have in mind (i.e., I can image wanting to reduce duplication), please let me know

Reply:
I think it makes sense to make this its own function and call it in both places even if we have to iterate twice. The list of listeners is unlikely to ever be long enough that it would make any significant performance difference, and it seems worth the tradeoff to make it more DRY

Reply:
Agreed, I'll change this.